sympy/printing/maple.py

```diff
@@ -0,0 +1,185 @@
+
+
+FIXME: This module is still under actively developed. Some functions may be not completed.
+"""
+
+from __future__ import print_function, division
+
+from sympy.codegen.ast import Assignment
+from sympy.core import S
+from sympy.core.compatibility import string_types, range
+from sympy.printing.codeprinter import CodePrinter
+from sympy.printing.precedence import precedence, PRECEDENCE
+
+_known_func_same_name = [
+    'sin', 'cos', 'tan', 'sec', 'csc', 'cot', 'sinh', 'cosh', 'tanh', 'sech',
+    'csch', 'coth', 'exp', 'floor'
+]
+
+known_functions = {
+    # Sympy -> Maple
+    # FIXME: maybe need added
+    'Abs': 'abs',
+    'log': 'ln',
+    'asin': 'arcsin',
+    'acos': 'arccos',
+    'atan': 'arctan',
+    'asec': 'arcsec',
+    'acsc': 'arccsc',
+    'acot': 'arccot',
+    'asinh': 'arcsinh',
+    'acosh': 'arccosh',
+    'atanh': 'arctanh',
+    'asech': 'arcsech',
+    'acsch': 'arccsch',
+    'acoth': 'arccoth',
+    'ceiling': 'ceil',
+}
+for _func in _known_func_same_name:
+    known_functions[_func] = _func
+
+atomic_expr = {
+    # Sympy -> Maple
+    S.Pi: 'Pi',
+    S.I: 'I',
+    S.Infinity: 'infinity'
+}
+
+spec_relational_ops = {
+    # Sympy -> Maple
+    '==': '=',
+    '!=': '<>'
+}
+
+
+class MapleCodePrinter(CodePrinter):
+    """
+    Printer which converts single sympy expressions into single.
+    """
+    def __init__(self, settings=None):
+        if settings is None:
+            settings = dict()
+        super(MapleCodePrinter, self).__init__(settings)
+        self.known_functions = dict(known_functions)
+        userfuncs = settings.get('user_functions', {})
+        self.known_functions.update(userfuncs)
+
+    def _get_statement(self, codestring):
+        return "%s;" % codestring
+
+    def _get_comment(self, text):
+        return "# {0}".format(text)
+
+    def _declare_number_const(self, name, value):
+        return "{0} := {1};".format(name,
+                                    value.evalf(self._settings['precision']))
+
+    def _format_code(self, lines):
+        return lines
+
+    def _print_Pow(self, expr):
+        PREC = precedence(expr)
+        if expr.exp == -1:
+            return '1/%s' % (self.parenthesize(expr.base, PREC))
+        elif expr.exp == 0.5 or expr.exp == S(1) / 2:
+            return 'sqrt(%s)' % self._print(expr.base)
+        elif expr.exp == -0.5 or expr.exp == -S(1) / 2:
+            return '1/sqrt(%s)' % self._print(expr.base)
+        else:
+            return '%s^%s' % (self._print(expr.base), self._print(expr.exp))
+
+    def _print_Rational(self, expr):
+        p, q = int(expr.p), int(expr.q)
+        return '%d/%d' % (p, q)
+
+    def _print_Relational(self, expr):
+        lhs_code = self._print(expr.lhs)
+        rhs_code = self._print(expr.rhs)
+        op = expr.rel_op
+        if op in spec_relational_ops:
+            op = spec_relational_ops[op]
+        return "{0} {1} {2}".format(lhs_code, op, rhs_code)
+
+    def _print_AtomicExpr(self, expr):
```

Comment:
After reading up a bit more, I think that maybe Number is a better class to print here. Or NumberSymbol (but oo etc is not NumberSymbol). AtomicExpr may include many other classes that you may be forced to handle as part of it (or add other print functions just to not get them in here). Sorry for not giving you fully correct advice to start with.

Reply:
And after seeing the test failures, maybe it is just the safest to go with _print_Infinity etc. Import/initialization orders are a mess to sort out... Sorry again.